### PR TITLE
feat: create uniform error structure

### PR DESCRIPTION
**Pull request by james-a-morris**

### CODE REVIEW

File: packages/error-handling/src/errors/IndexerHTTPError.ts [hunk 1]
```diff
@@ -10,22 +10,11 @@ export { StatusCodes };
...
-  /**
-   * A function used by `JSON.stringify` to specify which data will be serialized
-   * @returns A formatted JSON
-   */
-  public toJSON(): Record<string, unknown> {
-    return {
-      statusCode: this.httpStatusCode,
-      ...super.toJSON(),
-    };
-  }
```
Comment (amateima) [inline packages/error-handling/src/errors/IndexerHTTPError.ts:30]: OOC why was this removed?

File: packages/indexer-api/src/error-handler.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      : DEFAULT_STATUS;
+    res.status(httpStatus).json(err.toJSON());
+  } else if (isHttpError(err)) {
+    res.status(err.status ?? DEFAULT_STATUS).json({
+      message: err.message,
+      error: "NavigationError",
+    });
+  } else if (err instanceof StructError) {
+    res.status(StatusCodes.BAD_REQUEST).json({
+      error: "InputValidationError",
```
Comment (amateima) [inline packages/indexer-api/src/error-handler.ts:34]: We can simplify this to `ValidationError`

File: packages/error-handling/src/errors/IndexerHTTPError.ts (hunk 1 above)
Comment (james-a-morris) [reply packages/error-handling/src/errors/IndexerHTTPError.ts:30]: It would add the status code to the response code but this info is already exposed via the http status code that will get thrown.

File: packages/indexer-api/src/error-handler.ts (hunk 2 above)
Comment (james-a-morris) [reply packages/indexer-api/src/error-handler.ts]: https://github.com/across-protocol/indexer/pull/115/commits/d5d6c4da68894c50f5abfc787693f378176fc85d

**Review by amateima — APPROVED** [review]
